conformance/lang/expresssions/literals/byte_array_literal.balt

```diff
@@ -0,0 +1,498 @@
+    io:println(base16 `fd00000aa8660b5b010006acdc0100000101000100010000`); // @output [253,0,0,10,168,102,11,91,1,0,6,172,220,1,0,0,1,1,0,1,0,1,0,0]
+    io:println(base16 `48656C6C6F20576F726C64`); // @output [72,101,108,108,111,32,87,111,114,108,100]
+    io:println(base16 `dfa124DF`); // @output [223,161,36,223]
+}
+
+Test-Case: output
+Description: Test different ways of writing the same base16 byte array literal with spaces.
+Label: byte-array-literal
+
+public function main() {
+    byte[] byteArray = base16 `aaabcfccadafcd341a4bdfabcd8912df`;
+
+    io:println(byteArray == base16 `AAABCFCCADAFCD341A4BDFABCD8912DF`); // @output true
+    io:println(byteArray == base16 `aa ab cf cc ad af cd 34 1a 4b df ab cd 89 12 df`); // @output true
+    io:println(byteArray == base16 `aaabcfccad afcd34 1a4bdf abcd8912df`); // @output true
+    io:println(byteArray == base16 ` aa ab c fcc a   d af cd    3 4  1a4 bdf abcd8 91 2d f  `); // @output true
+}
+
+Test-Case: output
+Description: Test different ways of writing the same base16 byte array literal with tabs.
+Fail-Issue: ballerina-lang#32545
+Label: byte-array-literal
+
+public function main() {
+    byte[] byteArray = base16 `aaabcfccadafcd341a4bdfabcd8912df`;
+
+    io:println(byteArray == base16 `aa	ab	cf	cc	ad	af	cd	34	1a	4b	df	ab	cd	89	12	df`); // @output true
+    io:println(byteArray == base16 `aaabcfccad	afcd34	1a4bdf	abcd8912df`); // @output true
+    io:println(byteArray == base16 `	aa	ab	c	fcc	a	d	af	cd	3	4	1a4	bdf	abcd8	91	2d	f	`); // @output true
+}
+
+Test-Case: output
+Description: Test different ways of writing the same base16 byte array literal with newlines.
+Fail-Issue: ballerina-lang#32545
+Label: byte-array-literal
+
+public function main() {
+    byte[] byteArray1 = base16 `aaabcfccadafcd341a4bdfabcd8912df`;
+
+    byte[] byteArray2 = base16 `aaabcfccadafcd34
+    1a4bdfabcd8912df`;
+
+    byte[] byteArray3 = base16 `aaabcfccadafcd34
+
+    1a4bdfabcd8912df`;
+
+    byte[] byteArray4 = base16 `aaabcfcca
+                                dafcd341
+                                a4bd
+                                fabcd8912df`;
+
+    io:println(byteArray1); // @output [170,171,207,204,173,175,205,52,26,75,223,171,205,137,18,223]
+    io:println(byteArray1 == byteArray2); // @output true
+    io:println(byteArray1 == byteArray3); // @output true
+    io:println(byteArray1 == byteArray4); // @output true
+}
+
+Test-Case: output
+Description: Test inherent type of base16 byte array literal is byte[N],
+where N is the number of bytes encoded by the Base16Literal.
+Fail-Issue: ballerina-lang#32542
+Label: byte-array-literal, is-expr
+
+public function main() {
+    io:println(base16 `` is byte[0]); // @output true
+    io:println(base16 `aaabcfccadafcd341a4bdfabcd8912df` is byte[16]); // @output true
+}
+
+Test-Case: output
+Description: Test static type of base16 byte array literal is byte[N],
+where N is the number of bytes encoded by the Base16Literal.
```

Comment:
IMO, it's better, if we can keep the indentation same as the above line to improve the readability.

Reply:
+1. will update